### DISCUSSION TITLE
@method and @deprecated

### DISCUSSION THREAD

**Discussion by olvlvl** [phpDocumentor/phpDocumentor#3658] [Q&A]
Hi everyone,

How does one mark a magic method as deprecated? There's no mention of this in the documentation. I tried different combination in PhpStorm but nothing works:
- `@deprecated @method int count()`
- `@method @deprecated int count()`
- `@method int count() @deprecated`

That would be nice to implement because event magic methods get deprecated sometimes.

**Comment by mvriel** [comment]
Hi @olvlvl,

At current there is no way to mark a magic method as being deprecated; in the draft of PSR-5 (the new PHPDoc standard) there is a syntax to make that possible but that is not implemented in phpDocumentor yet.

The new Syntax would be:

```
@method int count() {
    @deprecated
}
```

**Comment by olvlvl** [comment]
Thanks for your answer @mvriel. I'll wait for PhpStorm to update then.

**Comment by midlan** [comment]
I've run into same problem today. So I guess it is not fixed yet, so we should reopen the issue then @olvlvl  ?

**Comment by olvlvl** [comment]
I completely forgot about this. I'm mostly writting Golang these days, no magic methods there :)

**Comment by midlan** [comment]
Ok I understand. Can you reopen the issue for us who miss that feature? Thanks.

**Comment by olvlvl** [comment]
@midlan Here you go.

**Comment by prodigeris** [comment]
Yeah, I'm missing this as well.

**Comment by dmitrach** [comment]
Also magic properties of classes get deprecated sometimes.
```php
/**
* @property int $magicProperty A value of the magic property. @deprecated v1.5 Will be removed.
*/
```
